play/client/src/main/scala/endpoints/play/client/Endpoints.scala

```diff
@@ -54,10 +54,16 @@ class Endpoints(host: String, wsClient: WSClient)(implicit val executionContext:
   type Response[A] = WSResponse => Either[Throwable, A]
 
   /** Successfully decodes no information from a response */
-  val emptyResponse: Response[Unit] = _ => Right(())
+  val emptyResponse: Response[Unit] = {
+    case resp if resp.status >= 200 && resp.status < 300 => Right(())
+    case resp => Left(new Throwable(s"Unexpected status code: ${resp.status}"))
+  }
 
   /** Successfully decodes string information from a response */
-  val textResponse: Response[String] = x => Right(x.body)
+  val textResponse: Response[String] = {
+    case resp if resp.status >= 200 && resp.status < 300 => Right(resp.body)
+    case resp => Left(new Throwable(s"Unexpected status code: ${resp.status}"))
+  }
```

Comment:
That’s probably out of the scope of this PR but it would really help to introduce a separate `ResponseEntity` type, so that we can separate filtering the response status and decoding the response entity.

Reply:
Yes, I had exactly the same thought. I even started to think about this but it seems harder than expected. E.g. `jsonResponse` right specifies(or should specify) 3 different things:
* ok response code(200 or maybe 2xx?)
* json encoded entity
* `Content-Type` header value

And this raises a question: how to implement basic algebra for defining e.g. response headers and status codes, so that `jsonRepsonse` can integrate with it easily(so preserve user-defined headers). 

Also much harder topic: how to implement status filtering? Should we allow to accept codes ranges(e.g. 2xx)? Should we introduce some kind of `or` operator and coproducts of payloads in such case? 

Maybe I will figure something out but don't have a good answer yet.

Reply:
I don’t have a good answer either!

Reply:
I have done it here :-) 

https://github.com/Fristi/endpoints-proto/blob/master/example/src/main/scala/itinere/example/ApiEndpoints.scala#L26